src/theory/quantifiers/sygus/sygus_interpol.h

```diff
@@ -0,0 +1,179 @@
+/*********************                                                        */
+/*! \file sygus_interpol.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Ying Sheng
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Sygus interpolation utility, which transforms an arbitrary input into
+ ** an interpolation problem.
+ **/
+
+#ifndef CVC4__THEORY__QUANTIFIERS__SYGUS_INTERPOL_H
+#define CVC4__THEORY__QUANTIFIERS__SYGUS_INTERPOL_H
+
+#include <string>
+#include <vector>
+
+#include "expr/node.h"
+#include "expr/type.h"
+#include "smt/smt_engine.h"
+
+namespace CVC4 {
+namespace theory {
+namespace quantifiers {
+/**
+ * A utility that turns a set of quantifier-free assertions into a sygus
+ * conjecture that encodes an interpolation problem. In detail, if our input
+ * formula is F( x ) for free symbols x, and is partitioned into axioms Fa and
+ * conjecture Fc then the sygus conjecture we construct is:
+ *
+ * exists A. forall x. ( (Fa( x ) => A( x )) ^ (A( x ) => Fc( x )) )
+ *
+ * where A( x ) is a predicate over the free symbols of our input that are
+ * shared between Fa and Fc. In other words, A( x ) must be implied by our
+ * axioms Fa and implies Fc( x ).
+ */
+class SygusInterpol
+{
+ public:
+  SygusInterpol();
+
+  SygusInterpol(LogicInfo logic);
+
+  /**
+   * Returns the sygus conjecture corresponding to the interpolation problem for
+   * input problem (F above) given by axioms (Fa above), and conj (Fc above).
+   *  Note that axioms is expected to be a subset of asserts.
+   *
+   * The argument name is the name for the interpol-to-synthesize.
+   *
+   * The type itpGType (if non-null) is a sygus datatype type that encodes the
+   * grammar that should be used for solutions of the interpolation conjecture.
+   *
+   * The output argument interpol is the solution to the synthesis conjector.
+   *
+   * The relationship between the free variables of asserts and the formal
+   * argument list of the interpol-to-synthesize are tracked by the attribute
+   * SygusVarToTermAttribute.
```

Comment:
Better describe what is the relationship and moreover why it is necessary.

Reply:
This is also copied from abduction, @ajreynol could you answer this question?

Reply:
Are you using this attribute?

Reply:
I tried to remove it, and it did passes the test cases. So I will remove it.